frontend/src/hooks/useWebSocketClient.ts

```diff
@@ -1,14 +1,18 @@
-import { io } from 'socket.io-client';
+import { io, type Socket } from 'socket.io-client';
 
-const webSocketClient = io(import.meta.env.VITE_WS_SERVER_URL);
+let webSocketClient: Socket | null = null;
 
 export function useWebSocketClient() {
+  if (!webSocketClient) {
+    webSocketClient = io(import.meta.env.VITE_WS_SERVER_URL);
+  }
+
```

Comment:
curiosity: what's this change for?

Reply:
This is introducing a singleton pattern that allowed to make sure we only initiate the client once, and also allowed to fix the related broken unit test at the same time.